File: server/src/shared/db/schema/venue.ts
import { integer, sqliteTable, text } from "drizzle-orm/sqlite-core";

export const venuesTable = sqliteTable("venue", {
  id: integer("id").primaryKey({ autoIncrement: true }),
  name: text("name").notNull(),
  city: text("city").notNull(),
  country: text("country").notNull(),
})

File: backend/cmd/cli/venue.go
package main

import (
	"context"
	"fmt"

	"github.com/jackc/pgx/v5"
)

func createVenueTable(ctx context.Context, tx pgx.Tx) error {
	query := `
	CREATE TABLE IF NOT EXISTS venue (
		id INTEGER GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
		name TEXT UNIQUE NOT NULL,
		city TEXT NOT NULL,
		country TEXT NOT NULL,

		created_at TIMESTAMP NOT NULL DEFAULT NOW(),
		updated_at TIMESTAMP NOT NULL DEFAULT NOW(),
		created_by INTEGER NOT NULL REFERENCES member(id),
		updated_by INTEGER NOT NULL REFERENCES member(id)
	);

	CREATE TRIGGER update_venue_timestamps BEFORE INSERT OR UPDATE ON venue
	FOR EACH ROW EXECUTE FUNCTION update_timestamps();`

	if _, err := tx.Exec(ctx, query); err != nil {
		return fmt.Errorf("Could not execute query: %v", err)
	}

	return nil
}
